parsl/tests/test_data/test_file_apps.py

```diff
@@ -1,74 +1,60 @@
-import os
-
 import pytest
 
-import parsl
 from parsl.app.app import bash_app
 from parsl.data_provider.files import File
-from parsl.tests.configs.local_threads import config
 
 
 @bash_app
-def cat(inputs=[], outputs=[], stdout=None, stderr=None):
-    infiles = ' '.join([i.filepath for i in inputs])
-    return """echo {i}
-    cat {i} &> {o}
-    """.format(i=infiles, o=outputs[0])
+def cat(inputs=(), outputs=(), stdout=None, stderr=None):
+    infiles = " ".join(i.filepath for i in inputs)
+    return f"cat {infiles} &> {outputs[0]}"
 
 
-@pytest.mark.usefixtures('setup_data')
 @pytest.mark.staging_required
-def test_files():
-
-    if os.path.exists('cat_out.txt'):
-        os.remove('cat_out.txt')
-
-    fs = [File('data/' + f) for f in os.listdir('data')]
-    x = cat(inputs=fs, outputs=[File('cat_out.txt')],
-            stdout='f_app.out', stderr='f_app.err')
+def test_files(setup_data):
+    fs = sorted(str(setup_data / f) for f in setup_data.iterdir())
+    fs = list(map(File, fs))
```

Comment:
does this need to be forced into being a list? or can it be left as a generator like object?
(I don't actually know the answer because the history of inputs= is that it was a list, and more recently my type checking work has pushed a lot of this stuff to be arbitrary `Sequence` objects... but all of that has been working on the assumption that what is supplied would be value-like (in the sense that you can observe multiple times... which you can't do with generator-like objects).
It's something to probably keep an eye on anywhere generator-like stuff gets passed into parsl

Reply:
I don't know.  When I first approached the set of test changes I was making these past two weeks, my onus was on keeping the spirit of what came before and "just" removing the painful `time.sleep()` calls.  In this case, "what came before" was:

```python
fs = [File('data/' + f) for f in os.listdir('data')]
```

which is definitely a list.  I'm happy to consider future Pythonic-focused changes, but I'm new enough to Parsl's code base that I'm currently being tepid.

Going forward, I'll keep your question in mind.